File: lib-content-scrapers/src/main/java/com/ustadmobile/lib/contentscrapers/ck12/practice/ScriptEngineReader.java
package com.ustadmobile.lib.contentscrapers.ck12.practice;

import java.io.File;
import java.io.FileReader;

import javax.script.Invocable;
import javax.script.ScriptEngine;
import javax.script.ScriptEngineManager;

public class ScriptEngineReader {

    private String result = "";

    public String getResult(String code) {

        try {

            FileReader cryptoReader = new FileReader(new File(getClass().getResource("/com/ustadmobile/lib/contentscrapers/ck12/crypto-js.js").toURI()));
            FileReader utils = new FileReader(new File(getClass().getResource("/com/ustadmobile/lib/contentscrapers/ck12/utils.js").toURI()));

            ScriptEngine engine = new ScriptEngineManager().getEngineByExtension("js");
            engine.eval(utils);
            engine.eval(cryptoReader);

            Invocable inv = (Invocable) engine;
            result = (String) inv.invokeFunction("getResult", code);

        } catch (Exception e) {
            e.printStackTrace();
        }

        return result;

    }
}

File: lib-content-scrapers/src/main/kotlin/com/ustadmobile/lib/contentscrapers/ck12/practice/ScriptEngineReader.kt
package com.ustadmobile.lib.contentscrapers.ck12.practice

import java.io.File
import java.io.FileReader

import javax.script.Invocable
import javax.script.ScriptEngine
import javax.script.ScriptEngineManager

class ScriptEngineReader {

    private var result = ""

    fun getResult(code: String): String {

        try {

            val cryptoReader = FileReader(File(javaClass.getResource("/com/ustadmobile/lib/contentscrapers/ck12/crypto-js.js").toURI()))
            val utils = FileReader(File(javaClass.getResource("/com/ustadmobile/lib/contentscrapers/ck12/utils.js").toURI()))

            val engine = ScriptEngineManager().getEngineByExtension("js")
            engine.eval(utils)
            engine.eval(cryptoReader)

            val inv = engine as Invocable
            result = inv.invokeFunction("getResult", code) as String

        } catch (e: Exception) {
            e.printStackTrace()
        }

        return result

    }
}
